src/components/Timeline/Event.js

```diff
@@ -76,10 +88,20 @@ Event.propTypes = {
 const eventHasValidDisplayText = (event) => event && event.data && event.data.DisplayText
 
 export const mapStateToEventProps = (state, ownProps) => {
-    const event = ownProps.event
-    const eventType = state.eventTypes.records[event.eventTypeId]
-    const ticket = state.tickets.map[ownProps.ticketId]
-    const engagement = state.engagements.list.find(engagement => engagement.id === ownProps.engagementId)
+  const event = ownProps.event
+  const eventType = state.eventTypes.records[event.eventTypeId]
+  const ticket = state.tickets.map[ownProps.ticketId]
+  const engagement = state.engagements.list.find(engagement => engagement.id === ownProps.engagementId)
```

Comment:
Event is not called with an engagementId. I know this isn't something you introduced, but it does cause a problem for determining which actions are qualified (for the ones that depend on the engagement). Hoisting the engagement selection to mapStateToDisplayPlaybookProps and passing engagementId alongside eventId, eventTypeId, ticketId, and incidentId though Playbook to DisplayPlaybook simplifies DisplayPlaybook significantly.